lisa/tools/iperf3.py

```diff
@@ -259,10 +261,12 @@ def run_as_client(
             client_ip,
             ip_version,
             udp_mode,
+            run_infinite,
         )
         return process.wait_result(
             expected_exit_code=0,
             expected_exit_code_failure_message="fail to launch iperf3 client",
+            timeout=run_time_seconds + 1,
```

Comment:
Many app use timeout=0 or -1 to control if it needs to wait or not. It needs only one variable to control the timeout. And it's weird to +1 here. Let's change to use 0 as infinite, and change the run_as_client_async to convert "0" to "inf".

Reply:
infinite may cause the whole running enters into dead loop?

Reply:
That should depend on how it's called, and the caller should fix it. If there are two arguments to control the same thing, it's easy to set the conflict variables, and get unexpected result.

Reply:
with iperf3 run with infinite, won't wait for the result, I update it into run_time_seconds

Reply:
If you check all place with infinite=False, the run_time_seconds are not 0. So it's safe to use 0 to present "infinite".

Reply:
in run_as_client, infinite set as False

Reply:
In the run_as_client, the run seconds are set to 10 seconds, it's the same as "infinite==false".

Reply:
current logic is 
```
        # set runtime to infinite if run_infinite is True
        if run_infinite:
            run_time = "inf"
        else:
            run_time = str(run_time_seconds)
```

Reply:
The condition can be changed to `if run_time_seconds<= 0:`

Reply:
how about this, I remove run_infinite parameter, set run_time_seconds as 0, when run_time_seconds is not passed or passed into a negative value, use run infinite.